NeoMathEngine/src/CPU/x86/CpuX86MathEngineVectorMathPrivate.h

```diff
@@ -873,11 +873,7 @@ inline void vectorMinMax( const float* first, float* result, const float minValu
 		for(int i = 0; i < sseSize; ++i) {
 			__m128 value = _mm_loadu_ps(first);
 
-			__m128 cmpMin = _mm_cmplt_ps(value, minSse);
-			__m128 cmpMax = _mm_cmpgt_ps(value, maxSse);
-			__m128 cmpNotNorm = _mm_or_ps(cmpMin, cmpMax);
-			__m128 res = _mm_or_ps(_mm_or_ps(_mm_andnot_ps(cmpNotNorm, value),
-				_mm_and_ps(cmpMin, minSse)), _mm_and_ps(cmpMax, maxSse));
+			__m128 res = _mm_min_ps(_mm_max_ps(value, minSse), maxSse);
```

Comment:
I've checked, this implementation loses `NaN` values (unlike previous one).

It could be an [issue](#576) because during training because during training the loss will be huge, but it will be a number. In previous implementation it's likely that `GetLastLoss()` will likely return `NaN` and it will give a little bit more information for figuring out the source of the problem (wrong net architecture/bug in preparing data).

Reply:
What about `NaN` in non SSE case?

Reply:
Good point! #578